### ISSUE TITLE
remove_entries method

### ISSUE THREAD

**Issue by michalstolarczyk** [pepkit/attmap#7]
I think it would be nice to have a `remove_entries` by key method since there is the `add_entries`. 
Not sure what would be the best way to specify the level of the entry ot remove. Relevant in case there are multiple keys with the same name on different levels.

**Comment by vreuter** [comment]
I agree -- I think there should be symmetry. I prefer to remove `add_entries` from the API, though, first by deprecation then by actual removal. The only things it's doing differently than an ordinary `update` call are type checking the entries argument, handling the dispatch to `__setitem__` to ensure that `Mapping` values become properly typed, and then returning then instance itself. 
This could all be accomplished instead with an overridden `update`, which--besides solving the odd asymmetry--would be more familiar to users since `update` is an ordinary `dict` operation.

**Comment by nsheff** [comment]
This is a fundamental design flaw of attmap — storing data in __dict__ means any key can shadow a method. Not fixable without rearchitecting the storage. Not worth doing at this point.
